src/Carbon/Traits/Creator.php

```diff
@@ -477,6 +477,37 @@ public static function createSafe($year = null, $month = null, $day = null, $hou
         return $instance;
     }
 
+    /**
+     * Create a new Carbon instance from a specific date and time using strict validation.
+     *
+     * @see create()
+     *
+     * @param int|null                 $year
+     * @param int|null                 $month
+     * @param int|null                 $day
+     * @param int|null                 $hour
+     * @param int|null                 $minute
+     * @param int|null                 $second
+     * @param DateTimeZone|string|null $tz
+     *
+     * @throws InvalidFormatException
+     *
+     * @return static
+     */
+    public static function createStrict(?int $year = 0, ?int $month = 1, ?int $day = 1, ?int $hour = 0, ?int $minute = 0, ?int $second = 0, $tz = null): self
+    {
+        $initialStrictMode = static::isStrictModeEnabled();
+        static::useStrictMode(true);
+
+        try {
+            $date = static::create($year, $month, $day, $hour, $minute, $second, $tz);
```

Comment:
Is there a point to using `createSafe()` here?

Reply:
```php
Carbon::useStrictMode(true);

echo Carbon::create(2021, 2, 30); // 2021-03-02
echo Carbon::createSafe(2021, 2, 30); // throw InvalidDateException Error: day : 30 is not a valid value.
```
Basically `createSafe` disallow overflow while `create()` actually allow 0 to 99 for any unit from month to second.

While `createSafe` is nice to prevent falling in unexpected result (as it ensure you either get a datetime with the numbers you pass or an exception in any other case), it's actually a bit slower as it makes the extra-checking around the `DateTime` constructor which allow overflow and it prevents convenient tricks like adding a fixed interval right in the `create()`:
```php
Carbon::create($year, $month, $day + 2); // $year-$month-$day + 2 days
```

Reply:
So - stick with the current code?

Reply:
So much regrets, please help me going back in time preventing myself adding this method.

We need to stick with `create()`, here are the blockers:
- `create()` is can also be used as an alias of `parse()`, e.g.: `Carbon::create('2021-01-02', 'Europe/London')`, stupid, but at some point judged convenient. And this is not allowed in `createSafe()`
- `createSafe()` doesn't allow overflow (such as February 30th overflowing to March 2nd) but default value for `$day` is `null` and `create()` use current value when passing `null`, convenient for time but a big mistake for the day, as mean `Carbon::createSafe(2020, 2)` actually throw an exception if today is a 29th, 30th or 31th. Actually this mistake is there since the very first release.

So rework needed for next major and I think `createSafe()` shouldn't be used internally by other methods meanwhile.